Tests/IntegrationTests/PSQLIntegrationTests.swift

```diff
@@ -360,4 +360,22 @@ final class IntegrationTests: XCTestCase {
         }
     }
     
+    func testConnectionClosureMidQueryDoesNotHang() async throws {
+        _ = await withThrowingTaskGroup(of: Void.self) { taskGroup in
+            for _ in (0 ..< 1_000) {
```

Comment:
why do we need to iterate a 1000 times here?

Reply:
Hmm not quite sure. I'll check to see if it makes an actual difference compared to like only 100.
The problem is that the test is a bit flaky.
The problem often does get triggered by this test, but once in like 4/5 runs it doesn't get triggered.

Reply:
The point that I want to make here, is that your test case does not seem to be deterministic. The example one I wrote below is deterministic. Let's focus on real unit tests where we mock out the db!

Reply:
I was wrong as mentioned below.
Changed the test.